trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -101,6 +105,78 @@ def readImagery() :
     # join the threads and complete
     workQueueData.join()
 
+
+def hdf5get(inputfile, key, metadata=False):
+    ''' Get a record from an hdf5 file '''
+    with h5py.File(inputfile, 'r') as f:
+        value = f[key]
+        if metadata:
+            rval = value[:], dict(value.attrs)
+        else:
+            rval = value[:]
+        return rval
+
+
+class FileData:

```

Comment:
The updates to openImage and the raw logic should largely make this factory logic obsolete. The walk method could be reduced to a lambda and the file size could be replaced with the np.ndarray shape. You will need to keep little logic to extract the image from the HDF5 file, however I believe the rest can be largely combined or keyed off other things common to both paths.